src/MapUpMap.js

```diff
@@ -32,18 +32,71 @@ function MapUpMap (props) {
 
   const toidLayer = {
     id: 'building-fill',
-    type: 'fill',
+    type: 'fill-extrusion',
     paint: {
-      'fill-color': '#3388ff',
-      'fill-opacity': 1
+      'fill-extrusion-color': '#0000FF',
+      'fill-extrusion-height': [
+        'interpolate',
+        ['linear'],
+        ['zoom'],
+        15,
+        0,
+        15.05,
+        heights ? ['get', 'relativeheightmaximum'] : 0
+      ]
+    }
+  }
+
+  const heightLayer = {
+    id: 'OS/TopographicArea_2/Building/1_3D',
+    type: 'fill-extrusion',
+    source: 'esri',
+    'source-layer': 'TopographicArea_2',
+    filter: ['all',
+      [
+        '==',
+        '_symbol',
+        4
+      ],
+      [
+        '!=',
```

Comment:
Given this, why not have two height layers, using the existing source, 1 which shows the highlighted building, and 1 which shows all other buildings?